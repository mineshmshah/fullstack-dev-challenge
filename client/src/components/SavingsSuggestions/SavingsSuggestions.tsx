import React, { FC, useContext } from 'react'
import { Text, Box, Grid, Image, Link } from '@chakra-ui/react'
import { ExternalLinkIcon } from '@chakra-ui/icons'
import { AppContext } from '../../store'
import { Card } from '../Card'

export const SavingsSuggestions: FC = () => {
    const { state } = useContext(AppContext)

    return (
        <Card>
            <Text
                fontSize='sm'
                marginBottom='4'
                color='blueHeader'
                fontWeight='700'
                textAlign='center'
            >
                Savings Suggestions
            </Text>

            <Text fontSize='xs' marginBottom='4' color='grey5' fontWeight='700' textAlign='center'>
                Here are some helpful articles just for your needs!
            </Text>
            <Box display='grid' gridTemplateColumns='1fr'>
                <Grid
                    width='auto'
                    padding='4'
                    gridTemplateColumns={{
                        base: '1fr',
                        md: 'repeat(2,1fr)',
                    }}
                    backgroundColor='blue200'
                    borderRadius='8px'
                >
                    {state.data.value && state.data.value?.finalValue > 50000 && (
                        <Grid
                            width='auto'
                            padding='4'
                            margin='4'
                            gridTemplateColumns={{
                                base: '1fr',
                                md: 'repeat(2,1fr)',
                            }}
                            backgroundColor='white'
                            borderRadius='8px'
                            justifyContent='center'
                            justifyItems='center'
                            alignItems='center'
                        >
                            <Image
                                src='https://www.finimize.com/wp/wp-content/uploads/2021/04/08042021-samsung-1.jpg'
                                alt='Segun Adebayo'
                                height='100px'
                            />
                            <Link
                                padding={4}
                                href='https://www.finimize.com/wp/news/smart-money-2/'
                                isExternal
                                textAlign='center'
                            >
                                Finimize: Smart Money <ExternalLinkIcon mx='2px' />
                            </Link>
                        </Grid>
                    )}
                    <Grid
                        width='auto'
                        padding='4'
                        margin='4'
                        gridTemplateColumns={{
                            base: '1fr',
                            md: 'repeat(2,1fr)',
                        }}
                        backgroundColor='white'
                        borderRadius='8px'
                        justifyContent='center'
                        justifyItems='center'
                        alignItems='center'
                    >
                        <Image
                            src='https://www.finimize.com/wp/wp-content/uploads/2021/03/06042021-emerging-markets.jpg'
                            alt='Segun Adebayo'
                            height='100px'
                        />
                        <Link
                            padding={4}
                            href='https://www.finimize.com/wp/news/now-youre-cookin/'
                            isExternal
                            textAlign='center'
                        >
                            Finimize: Now You’re Cookin’ <ExternalLinkIcon mx='2px' />
                        </Link>
                    </Grid>
                </Grid>
            </Box>
        </Card>
    )
}
