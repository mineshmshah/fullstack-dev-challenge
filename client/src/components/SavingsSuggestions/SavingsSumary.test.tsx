import React from 'react'
import { cleanup } from '@testing-library/react'
import { customRender } from '../../utils/test-utils'
import { initialState } from '../../store/initialState'
import { SavingsSuggestions } from './SavingsSuggestions'

afterEach(cleanup)

describe('component/SavingsSuggestion', () => {
    it('renders Savings suggestion component', () => {
        const { render } = customRender(<SavingsSuggestions />, {})
        const { container } = render
        expect(container).toMatchSnapshot()
    })
    it('renders Savings suggestion component with high value savings', () => {
        const { render } = customRender(<SavingsSuggestions />, {
            store: {
                ...initialState,
                data: {
                    ...initialState.data,
                    value: {
                        finalValue: 50001,
                        yearlySavings: [],
                        yearlyBreakdown: [],
                    },
                },
            },
        })
        const { container } = render
        expect(container).toMatchSnapshot()
    })
})
