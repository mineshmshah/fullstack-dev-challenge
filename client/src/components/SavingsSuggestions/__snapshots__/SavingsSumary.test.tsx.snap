// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/SavingsSuggestion renders Savings suggestion component 1`] = `
<div>
   
  <div
    class="css-6z87o"
  >
    <p
      class="chakra-text css-nrvl4t"
    >
      Savings Suggestions
    </p>
    <p
      class="chakra-text css-8w3mbw"
    >
      Here are some helpful articles just for your needs!
    </p>
    <div
      class="css-12e5m3r"
    >
      <div
        class="css-q6o7lv"
      >
        <div
          class="css-1p3ap9t"
        >
          <img
            alt="Segun Adebayo"
            class="chakra-image__placeholder css-73nay0"
          />
          <a
            class="chakra-link css-1m2xxl"
            href="https://www.finimize.com/wp/news/now-youre-cookin/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Finimize: Now You’re Cookin’ 
            <svg
              class="chakra-icon css-q5nc5y"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <path
                  d="M15 3h6v6"
                />
                <path
                  d="M10 14L21 3"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
   
   
</div>
`;

exports[`component/SavingsSuggestion renders Savings suggestion component with high value savings 1`] = `
<div>
   
  <div
    class="css-6z87o"
  >
    <p
      class="chakra-text css-nrvl4t"
    >
      Savings Suggestions
    </p>
    <p
      class="chakra-text css-8w3mbw"
    >
      Here are some helpful articles just for your needs!
    </p>
    <div
      class="css-12e5m3r"
    >
      <div
        class="css-q6o7lv"
      >
        <div
          class="css-1p3ap9t"
        >
          <img
            alt="Segun Adebayo"
            class="chakra-image__placeholder css-73nay0"
          />
          <a
            class="chakra-link css-1m2xxl"
            href="https://www.finimize.com/wp/news/smart-money-2/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Finimize: Smart Money 
            <svg
              class="chakra-icon css-q5nc5y"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <path
                  d="M15 3h6v6"
                />
                <path
                  d="M10 14L21 3"
                />
              </g>
            </svg>
          </a>
        </div>
        <div
          class="css-1p3ap9t"
        >
          <img
            alt="Segun Adebayo"
            class="chakra-image__placeholder css-73nay0"
          />
          <a
            class="chakra-link css-1m2xxl"
            href="https://www.finimize.com/wp/news/now-youre-cookin/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Finimize: Now You’re Cookin’ 
            <svg
              class="chakra-icon css-q5nc5y"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <path
                  d="M15 3h6v6"
                />
                <path
                  d="M10 14L21 3"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
   
   
</div>
`;
